### PR TITLE
Doc: Navigation fix and bump mkdocs-material version

### PR DESCRIPTION
**Pull request by JulioPDX**
## Change Summary

This update fixes the issue introduced by the `navigation.indexes` feature. This PR also migrates to leveraging the mkdocs container by [squidfunk](https://hub.docker.com/r/squidfunk/mkdocs-material)

## Proposed changes

- Disable `navigation.indexes` feature
- Update footer for 2023
- Remove unused requirement (mkdocs-bootswatch)
- Bump Mkdocs-material version from `8.5.7` to `9.0.12`
- Enable `content.code.copy` feature, must be specified when moving to version 9.x
- Update docker compose container
- Update container check from host vs within container

## How to test

Either build locally or reference example [site](https://ansible-avd-juliopdx.readthedocs.io/en/doc-fixes/index.html).

- Checkout this PR
- `python3 -m venv venv`
- `source venv/bin/activate`
- `pip3 install -r ansible_collections/arista/avd/docs/requirements.txt`
- `mkdocs serve`

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: .github/workflows/pull-request-management.yml [hunk 1]
```diff
@@ -103,7 +103,7 @@ jobs:
           docker-compose -f docker-compose.yml ps
       - name: 'Test connectivity to mkdoc server'
         run: |
-          until docker exec webdoc_avd curl -s -I http://localhost:8000/ ; do sleep 2; done
+          bash -c 'while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' 127.0.0.1:8000)" != "200" ]]; do sleep 5; done'
```
Comment (ClausHolbechArista) [inline .github/workflows/pull-request-management.yml:107]: Should we add a timeout on this to avoid the default 12-hour timeout for the job?

Comment (ClausHolbechArista) [reply .github/workflows/pull-request-management.yml:107]: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepstimeout-minutes

Comment (JulioPDX) [reply .github/workflows/pull-request-management.yml:107]: ill get a timeout added to that step. thank you

Comment (JulioPDX) [reply .github/workflows/pull-request-management.yml:107]: considering the length of our ci, thoughts on doing something like 40 mins?

Comment (JulioPDX) [reply .github/workflows/pull-request-management.yml:107]: disregard for that step ill do something much shorter

Comment (JulioPDX) [reply .github/workflows/pull-request-management.yml:107]: set for 10 minutes, pipeline usually completes in 5 mins

File: development/docker-compose.yml [hunk 2]
```diff
@@ -15,21 +15,21 @@ services:
...
-    image: titom73/mkdocs:latest
+    image: squidfunk/mkdocs-material:9.0.12
     container_name: webdoc_avd
     volumes:
       - ${PWD}/:/docs
     ports:
       - 8000:8000
     entrypoint: ""
-    command: ["sh", "-c", "cd ansible-avd/ && pip install -r ansible_collections/arista/avd/docs/requirements.txt && mkdocs serve --no-livereload --dev-addr=0.0.0.0:8000 -f mkdocs.yml"]
+    command: ["sh", "-c", "cd ansible-avd/ && pip install -r ansible_collections/arista/avd/docs/requirements.txt --upgrade && mkdocs serve --no-livereload --dev-addr=0.0.0.0:8000 -f mkdocs.yml"]
```
Comment (gmuloc) [inline development/docker-compose.yml:35]: so maybe not for this PR ... I would challenge keeping this if we change the container. if you look at our CI pipeline we are actually doing some `sed` operation to remove the `cd ansible-avd`.

https://github.com/aristanetworks/ansible-avd/pull/2536/files#diff-6f0e9ef0c3a211590d88765ce7dee35c551f602cf76d5d67fe66828b6c34ae3cR102

A more general question is do we need to keep docker-compose at this stage - is anybody still using this docker-compose ?

Maybe we could start webdoc_avd as a service container for the job instead
https://docs.github.com/en/actions/using-containerized-services/about-service-containers

**Review by ClausHolbechArista — APPROVED** [review]

**Review by gmuloc — APPROVED** [review]

**Pull request merged**